snuba/subscriptions/entity_subscription.py

```diff
@@ -0,0 +1,122 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, List, Mapping, Optional, Sequence, Type
+
+from snuba.datasets.entities import EntityKey
+from snuba.query.conditions import ConditionFunctions, binary_condition
+from snuba.query.exceptions import InvalidQueryException
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.types import Condition
+
+
+class SubscriptionType(Enum):
+    LEGACY = "legacy"
+    SNQL = "snql"
+    DELEGATE = "delegate"
+
+
+class EntitySubscription(ABC):
+    def __init__(self, data_dict: Mapping[str, Any]) -> None:
+        ...
```

Comment:
This isn't necessary

Reply:
It actually is though, because I will get an error if I do for example `EventsSubscription(data_dict={})` because it will say unexpected keyword